### ISSUE TITLE
[bug] wildcards in profiles for env variables do not work

### ISSUE THREAD

**Issue by scddev** [conan-io/conan#8657]
### Environment Details
  * Ubuntu 20.04
  * gcc/clang
  * 1.33.1
  * 3.8

### Steps to reproduce

E.g. enhance the sample for profiles from the docs https://docs.conan.io/en/latest/reference/profiles.html:

```
[settings]
*@myuser/*:compiler=clang
*@myuser/*:compiler.version=3.5
*@myuser/*:compiler.libcxx=libstdc++11
compiler=gcc
compiler.version=4.9
compiler.libcxx=libstdc++11

[env]
*@myuser/*:CC=/usr/bin/clang
*@myuser/*:CXX=/usr/bin/clang++
CC=/usr/bin/gcc
CXX=/usr/bin/g++
```

If repeating the env variables with every package name (like in docs with e.g. zlib), it works as expected.

**Comment by memsharded** [comment]
Proposed a potential fix in https://github.com/conan-io/conan/pull/8688 for next 1.35

**Comment by memsharded** [comment]
Merged https://github.com/conan-io/conan/pull/8688, this will be released in 1.35
